prestashopv1/plugin/modules/indodana/tools/IndodanaTools.php

```diff
@@ -0,0 +1,166 @@
+                'seller' => self::getSeller()
+            ]);
+        }
+
+        return self::$indodanaCommon;
+    }
+
+    public static function getSeller()
+    {
+        return [
+            'name' => Configuration::get('INDODANA_STORE_NAME'),
+            'email' => Configuration::get('INDODANA_STORE_EMAIL'),
+            'url' => Configuration::get('INDODANA_STORE_URL'),
+            'address' => [
+                'firstName' => Configuration::get('INDODANA_STORE_NAME'),
+                'lastName' => '',
+                'address' => Configuration::get('INDODANA_STORE_ADDRESS'),
+                'city' => Configuration::get('INDODANA_STORE_CITY'),
+                'postalCode' => Configuration::get('INDODANA_STORE_POSTAL_CODE'),
+                'phone' => Configuration::get('INDODANA_STORE_PHONE'),
+                'countryCode' => Configuration::get('INDODANA_STORE_COUNTRY_CODE')
+            ]
+        ];
+    }
+
+    public static function getTotal($cart)
+    {
+        return $cart->getOrderTotal(true, Cart::BOTH);
+    }
+
+    public static function getShippingFee($cart)
+    {
+        return $cart->getOrderTotal(true, Cart::ONLY_SHIPPING);
+    }
+
+    public static function getDiscount($cart)
+    {
+        $discount = 0;
+        $products = $cart->getProducts();
+        foreach ($products as $product) {
+            if (_PS_VERSION_ >= 1.7) {
+                $discount += $product['reduction'] * $product['quantity'];
+            } else {
+                $discount += ($product['price_without_reduction'] - $product['price_with_reduction'])
+                    * $product['quantity'];
+            }
+        }
+
+        return round($discount);
+    }
+
+    public static function getTax($cart)
+    {
+        $tax = 0;
+        $deficit = 0;
+        $products = $cart->getProducts();
+        foreach ($products as $product) {
+            $tax += ($product['price_without_reduction'] - self::getPriceWithoutReductionWithoutTax($product))
+                * $product['quantity'];
+            $deficit += self::getDeficit($product);
+        }
+
+        return round($tax) + $deficit;
+    }
+
+    public static function getProducts($cart)
+    {
+        $items = [];
+        $products = $cart->getProducts();
+        foreach ($products as $product) {
+            $items[] = [
+                'id' => $product['reference'],
+                'name' => $product['name'],
+                'price' => round(self::getPriceWithoutReductionWithoutTax($product)),
+                'quantity' => $product['quantity'],
+                'category'  => IndodanaCommon\IndodanaConstant::DEFAULT_ITEM_CATEGORY,
+            ];
+        }
+
+        return $items;
+    }
+
+    public static function getCustomerDetails($orderId)
+    {
+        $orderDetails = new Order((int) $orderId);
+        $customer = new Customer((int) $orderDetails->id_customer);
+        $invoiceDetails = new Address((int) $orderDetails->id_address_invoice);
```

Comment:
Hmm I'm not sure about this one, but shouldn't `new` not query from database? In other words, there should be a query layer instead of querying directly from constructor right? Open for discussion ya. For example 
```php
$orderQuery = new OrderQuery($conn)
$order = $orderQuery->fetchById($orderId);
```

Reply:
`Order` is Prestashop ObjectModel class and `new Order((int) $orderId)` is to fetch order details data.
I know that's unusual, but it's Prestashop's way to fetch data from database. other prestashop modules did the same too.

Reply:
Alright, I agree, it's better to follow platform's convention, thank you for mentioning this.